### PR TITLE
Integrate DBW as "Best Practices" into main report

### PR DESCRIPTION
**Pull request by ebidel**
R: @paulirish @GoogleChrome/lighthouse 

The adds a new (and hopefully temp) default_and_dbw.json with the dbw audits mixed into the default.json. This file is now used for the default LH run. The plan is that this is temporary until we figure out next steps for https://github.com/GoogleChrome/lighthouse/issues/887. But it should be transparent to users. Developers will want to add new core audits to default_and_dbw.json and either default.js (core audit) or dobetterweb.json  (if it's dbw).

For default_and_dbw.json, I've:
- given dbw passes a `passName: dbw`.
- Put DBW under "Best Practices". The list is "PWA, BP, Perf Metrics".
- moved A11y under "Best Practices"

FWIW, The "Technology" view is a little funky right now. I didn't want to touch that until the new report is done.

Wee!

screen shot 2016-11-06 at 12 41 25 pm

### CODE REVIEW

File: lighthouse-core/config/default_and_dbw.json [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+  {
+    "gatherers": [
+      "http-redirect",
+      "html-without-javascript"
+    ]
+  }, {
+    "recordNetwork": true,
+    "passName": "dbw",
+    "gatherers": [
+      "https",
```
Comment (brendankenny) [inline lighthouse-core/config/default_and_dbw.json]: duplicate `https` and `url` gatherers here

File: lighthouse-core/config/default_and_dbw.json [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+      "styles",
+      "url",
+      "./gather/gatherers/dobetterweb/appcache",
```
Comment (brendankenny) [inline lighthouse-core/config/default_and_dbw.json]: now that you've expanded `getAuditList` and `getGathererList` to include `dobetterweb/`, all of these can drop the path and just start with the directory (e.g. `"dobetterweb/appcache"`)

File: lighthouse-core/runner.js [hunk 3]
```diff
@@ -142,19 +142,24 @@ class Runner {
    * @return {!Array<string>}
    */
   static getAuditList() {
-    return fs
-        .readdirSync(path.join(__dirname, './audits'))
-        .filter(f => /\.js$/.test(f) && f !== 'audit.js');
+    const fileList = [
+      ...fs.readdirSync(path.join(__dirname, './audits')),
+      ...fs.readdirSync(path.join(__dirname, './audits/dobetterweb')).map(i => `dobetterweb/${i}`)
```
Comment (brendankenny) [inline lighthouse-core/runner.js]: nit: use `f` instead of `i`

File: lighthouse-core/runner.js [hunk 4]
```diff
@@ -142,19 +142,24 @@ class Runner {
...
    * @return {!Array<string>}
    */
   static getGathererList() {
-    return fs
-        .readdirSync(path.join(__dirname, './gather/gatherers'))
-        .filter(f => /\.js$/.test(f) && f !== 'gatherer.js');
+    const fileList = [
+      ...fs.readdirSync(path.join(__dirname, './gather/gatherers')),
+      ...fs.readdirSync(path.join(__dirname, './gather/gatherers/dobetterweb'))
+          .map(i => `dobetterweb/${i}`)
```
Comment (brendankenny) [inline lighthouse-core/runner.js]: nit: `f` instead of `i`

File: lighthouse-core/test/runner-test.js [hunk 5]
```diff
@@ -245,7 +245,7 @@ describe('Runner', () => {
     const coreAudits = Runner.getAuditList();
     coreAudits.forEach(auditFilename => {
       const auditPath = '../audits/' + auditFilename;
-      const auditExpectedName = /([^.]+)\.js$/.exec(auditFilename)[1];
+      const auditExpectedName = /(dobetterweb\/)?([^.]+)\.js$/.exec(auditFilename)[2];
```
Comment (brendankenny) [inline lighthouse-core/test/runner-test.js]: we should probably just switch to `path.basename(auditFilename, '.js')`

File: lighthouse-core/config/default_and_dbw.json (hunk 1 above)
Comment (ebidel) [inline lighthouse-core/config/default_and_dbw.json]: done

File: lighthouse-core/runner.js (hunk 4 above)
Comment (ebidel) [inline lighthouse-core/runner.js]: done

File: lighthouse-core/runner.js (hunk 3 above)
Comment (ebidel) [inline lighthouse-core/runner.js]: done

File: lighthouse-core/test/runner-test.js (hunk 5 above)
Comment (ebidel) [inline lighthouse-core/test/runner-test.js]: done

File: lighthouse-core/config/default_and_dbw.json [hunk 6]
```diff
@@ -0,0 +1,412 @@
+{
```
Comment (ebidel) [inline lighthouse-core/config/default_and_dbw.json]: done

File: lighthouse-core/config/default.json [hunk 7]
```diff
@@ -25,6 +25,22 @@
       "http-redirect",
       "html-without-javascript"
     ]
+  }, {
+    "recordNetwork": true,
+    "passName": "dbw",
+    "gatherers": [
+      "styles",
+      "url",
+      "./gather/gatherers/dobetterweb/anchors-with-no-rel-noopener",
```
Comment (brendankenny) [inline lighthouse-core/config/default.json]: just
```js
"dobetterweb/anchors-with-no-rel-noopener",
"dobetterweb/appcache",
"dobetterweb/console-time-usage",
"dobetterweb/datenow",
```
etc

File: lighthouse-core/config/default.json [hunk 8]
```diff
@@ -58,7 +74,19 @@
     "image-alt",
     "label",
     "tabindex",
-    "content-width"
+    "content-width",
+    "./audits/dobetterweb/external-anchors-use-rel-noopener",
```
Comment (brendankenny) [inline lighthouse-core/config/default.json]:
```js
"dobetterweb/external-anchors-use-rel-noopener",
"dobetterweb/appcache-manifest",
"dobetterweb/geolocation-on-start",
```
etc

File: lighthouse-core/config/default.json [hunk 9]
```diff
@@ -25,6 +25,22 @@
       "http-redirect",
       "html-without-javascript"
     ]
+  }, {
+    "recordNetwork": true,
+    "passName": "dbw",
+    "gatherers": [
+      "styles",
+      "url",
```
Comment (brendankenny) [inline lighthouse-core/config/default.json]: remove `url` (runs in first pass)

File: package.json [hunk 10]
```diff
@@ -22,7 +22,7 @@
     "closure": "cd lighthouse-core && closure/closure-type-checking.js",
     "watch": "lighthouse-core/scripts/run-mocha.sh --watch",
     "chrome": "lighthouse-core/scripts/launch-chrome.sh",
-    "dbw": "npm run start -- --config-path=lighthouse-core/config/dobetterweb.json --disable-device-emulation --disable-cpu-throttling --disable-network-throttling --output=html --output-path=results.html",
+    "dbw": "npm run start -- --disable-device-emulation --disable-cpu-throttling --disable-network-throttling --output=html --output-path=results.html",
```
Comment (brendankenny) [inline package.json]: if we're going to get rid of the dbw-only config, this should be dropped (default already covers everything here). If it's just for the dropping of emulation, we could rename this `fast` or something

File: lighthouse-core/config/default_and_dbw.json (hunk 2 above)
Comment (ebidel) [inline lighthouse-core/config/default_and_dbw.json]: done

File: lighthouse-core/config/default.json (hunk 8 above)
Comment (ebidel) [inline lighthouse-core/config/default.json]: done

File: lighthouse-core/config/default.json (hunk 7 above)
Comment (ebidel) [inline lighthouse-core/config/default.json]: done

File: lighthouse-core/config/default.json (hunk 9 above)
Comment (ebidel) [inline lighthouse-core/config/default.json]: done

File: package.json (hunk 10 above)
Comment (ebidel) [inline package.json]: done

**Comment by brendankenny** [comment]
I've preemptively squashed as appropriate to keep config vs smoke test changes separate :P

**Pull request merged**